h2c/server.go

```diff
@@ -44,12 +52,46 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+	s.trackConn(conn, true)
+	defer s.trackConn(conn, false)
 
 	new(http2.Server).ServeConn(bufConn{conn, bufrw}, &http2.ServeConnOpts{
 		Handler: s.HTTP2Handler,
 	})
 }
 
+func (s *Server) Shutdown(ctx context.Context) error {
+	ticker := time.NewTicker(shutdownPollInterval)
+	defer ticker.Stop()
+	for {
+		// TODO: This is a pretty naive approach.
+		// Just checking whether there are any connections used for requests
+		// that haven't yet finished. Hopefully it's "good enough".
+		if len(s.connections) == 0 {
```

Comment:
I don't think this actually matters, but this is technically an unsynchronized read. Might be worth throwing a rmutex lock/unlock in here.

Reply:
+1